src/CommunityToolkit.Maui.Core/Essentials/FileFolderDialog.tizen.cs

```diff
@@ -0,0 +1,398 @@
+			else
+			{
+				SendSubmit(selectedPath);
+			}
+		};
+		hlayout.Add(okButton);
+
+		Relayout += (s, e) =>
+		{
+			var isHorizontal = Window.Instance.WindowSize.Width > Window.Instance.WindowSize.Height;
+			content.SizeWidth = Window.Instance.WindowSize.Width * (isHorizontal ? 0.5f : 0.8f);
+		};
+
+		return content;
+	}
+
+	static bool CreateSubDirectory(string newDirectory)
+	{
+		Directory.CreateDirectory(newDirectory);
+		return Directory.Exists(newDirectory);
+	}
+
+	void ProcessSelect(string selectedItem)
+	{
+		if (selectedItem.Equals(previousDirectorySymbol, StringComparison.Ordinal))
+		{
+			if (selectedPath.Equals(initialFileFolderPath, StringComparison.Ordinal)
+				|| (selectedPath + slashSymbol).Equals(initialFileFolderPath, StringComparison.Ordinal))
+			{
+				return;
+			}
+			selectedPath = Path.GetDirectoryName(selectedPath) ?? initialFileFolderPath;
+			selectedPath = selectedPath[..selectedPath.LastIndexOf(slashSymbol, StringComparison.Ordinal)];
+		}
+		else
+		{
+			if (!IsDirectory(selectedPath))
+			{
+				selectedPath = Path.GetDirectoryName(selectedPath) ?? selectedPath[..selectedPath.LastIndexOf(slashSymbol, StringComparison.Ordinal)];
+			}
+			selectedPath = selectedPath.EndsWith(slashSymbol) ? selectedPath : selectedPath + slashSymbol;
+			selectedPath += selectedItem;
+		}
+
+		if (IsDirectory(selectedPath))
+		{
+			UpdateDirectoryScrollView(selectedPath);
+		}
+		else if (isFileSelectionMode)
+		{
+			fileNameEntry!.Text = selectedItem;
+		}
+	}
+
+	void UpdateDirectoryScrollView(string path)
+	{
+		if (directoryScrollView is null)
+		{
+			return;
+		}
+
+		foreach(var view in directoyViews)
+		{
+			directoryScrollView.ContentContainer.Remove(view);
+		}
+		directoyViews.Clear();
+
+		var listItems = GetDirectories(path);
+
+		if (listItems is not null)
+		{
+			foreach (var item in listItems)
+			{
+				var itemLabel = new Label
+				{
+					Text = item,
+					Focusable = true,
+					HorizontalTextAlignment = Tizen.UIExtensions.Common.TextAlignment.Start,
+					PixelSize = 16d.ToPixel(),
+					WidthSpecification = LayoutParamPolicies.MatchParent,
+					HeightSpecification = LayoutParamPolicies.WrapContent,
+					Margin = new Extents(0, 0, (ushort)5d.ToPixel(), (ushort)5d.ToPixel()),
+				};
+				itemLabel.TouchEvent += (s, e) =>
+				{
+					var state = e.Touch.GetState(0);
+					if (state == PointStateType.Up && itemLabel.IsInside(e.Touch.GetLocalPosition(0)))
+					{
+						ProcessSelect(item);
+						return true;
+					}
+					return false;
+				};
+				itemLabel.KeyEvent += (s, e) =>
+				{
+					if (e.Key.IsAcceptKeyEvent())
+					{
+						ProcessSelect(item);
+						return true;
+					}
+					return false;
+				};
+				directoryScrollView.ContentContainer.Add(itemLabel);
+				directoyViews.Add(itemLabel);
+			}
+			directoryScrollView.SizeHeight = 30d.ToPixel() * Math.Min(listItems.Count(), 5);
+		}
+	}
+
+	IEnumerable<string> GetDirectories(string path)
+	{
+		var directories = new List<string>();
```

Comment:
instead of creating a collection, can we use the `yield return`?

Reply:
Yes, I've wanted directories to be sorted, but I don't see it is necessary now.